### ISSUE TITLE
ModuleNotFoundError: No module named 'zlib_ng'

### ISSUE THREAD

**Issue by tjorim** [sca075/valetudo_vacuum_camera#126]
What version of an integration has described problem?
2024.04.4

What was the last working version of an integration?
n/a

What version of Home Assistant do you use?
core-2024.5.0b0

```
This error originated from a custom integration.

Logger: homeassistant
Source: custom_components/valetudo_vacuum_camera/valetudo/MQTT/connector.py:14
integration: Valetudo Vacuum Camera (documentation, issues)
First occurred: April 24, 2024 at 23:04:40 (1 occurrences)
Last logged: April 24, 2024 at 23:04:40

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1952, in async_forward_entry_setup
    return await self._async_forward_entry_setup(entry, domain, True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1975, in _async_forward_entry_setup
    await integration.async_get_platform(domain)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1069, in async_get_platform
    platforms = await self.async_get_platforms((platform_name,))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1146, in async_get_platforms
    import_future.result()
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1116, in async_get_platforms
    await self.hass.async_add_import_executor_job(
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1059, in _load_platforms
    platform_name: self._load_platform(platform_name)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1233, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1265, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 144, in protected_loop_func
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/valetudo_vacuum_camera/camera.py", line 114, in <module>
    from .valetudo.MQTT.connector import ValetudoConnector
  File "/config/custom_components/valetudo_vacuum_camera/valetudo/MQTT/connector.py", line 14, in <module>
    from zlib_ng import gzip_ng as gzip
ModuleNotFoundError: No module named 'zlib_ng'
```

_Originally posted by @tjorim in https://github.com/sca075/valetudo_vacuum_camera/issues/124#issuecomment-2076493049_

**Created new issue as it's different from the deprecated alias.**

**Comment by tjorim** [comment]
https://github.com/sca075/valetudo_vacuum_camera/issues/124#issuecomment-2076347600

> It is the zlib_ng implemented in the connector.py will work on it later today.
> 
> https://github.com/sca075/valetudo_vacuum_camera/blob/d8a090b0d7a4c5b8f75ea87418bab8ba208555f1/custom_components/valetudo_vacuum_camera/valetudo/MQTT/connector.py#L14-L15

https://github.com/sca075/valetudo_vacuum_camera/issues/124#issuecomment-2076792015

> Changes related to the `zlib_ng`: [home-assistant/core#115767](https://github.com/home-assistant/core/pull/115767), and afterwards [home-assistant/core#115777](https://github.com/home-assistant/core/pull/115777).

https://github.com/sca075/valetudo_vacuum_camera/issues/124#issuecomment-2076914409

> I thought it would be as simple as changing those lines from
> 
> ```
>  from zlib_ng import gzip_ng as gzip
>  from zlib_ng import zlib_ng as zlib
> ```
> 
> to
> 
> ```
> from isal import igzip as gzip
> from isal import isal_zlib as zlib
> ```
> 
> but now getting this error:
> 
> ```
> Logger: homeassistant.util.logging
> Source: util/logging.py:98
> First occurred: 12:55:18 (2 occurrences)
> Last logged: 12:55:44
> 
> Exception in async_message_received when handling msg on 'valetudo/Alfred/BatteryStateAttribute/level': '99' Traceback (most recent call last): File "/config/custom_components/valetudo_vacuum_camera/valetudo/MQTT/connector.py", line 322, in async_message_received await self.hypfer_handle_battery_level(msg) File "/config/custom_components/valetudo_vacuum_camera/valetudo/MQTT/connector.py", line 208, in hypfer_handle_battery_level self._mqtt_vac_battery_level = int(bytes.decode(self._payload, "utf-8")) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ TypeError: descriptor 'decode' for 'bytes' objects doesn't apply to a 'str' object
> Exception in async_message_received when handling msg on 'valetudo/Alfred/BatteryStateAttribute/level': '98' Traceback (most recent call last): File "/config/custom_components/valetudo_vacuum_camera/valetudo/MQTT/connector.py", line 322, in async_message_received await self.hypfer_handle_battery_level(msg) File "/config/custom_components/valetudo_vacuum_camera/valetudo/MQTT/connector.py", line 208, in hypfer_handle_battery_level self._mqtt_vac_battery_level = int(bytes.decode(self._payload, "utf-8")) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ TypeError: descriptor 'decode' for 'bytes' objects doesn't apply to a 'str' object
> ```

**Comment by sca075** [comment]
@tjorim 
Please replace the lines in the connector.py with a simple 
import gzip
import zlib 

This solves the issue;)

**Comment by sca075** [comment]
@tjorim  done on the main, will publish the fix after a short test.

**Comment by tjorim** [comment]
Working well for me.
Closed by https://github.com/sca075/valetudo_vacuum_camera/commit/c0718c506a973ac97ff43d3bfaa97f20311e7ba0.

**Comment by sca075** [comment]
@tjorim I made some additional test with the isal, and I got the same error you had with the '''BatteryStateAttribute'' the issue was in the bytes decode that of course can be done automatically by simply converting the bytes to _int_ when I will release the new version of the camera in the connector.py we can use the isal library (that performance wise is faster then gzip and zlib python embed libs.. of course there is the need to do more tests on it.
Once again thank you really much for the help and you are welcome to contribute at any time. I do really appreciate your kind cooperation.
